src/test/java/org/gridsuite/shortcircuit/server/ShortCircuitParametersControllerTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.shortcircuit.server;
+
+import lombok.NonNull;
+import org.assertj.core.api.WithAssertions;
+import org.gridsuite.shortcircuit.server.dto.ShortCircuitParametersInfos;
+import org.gridsuite.shortcircuit.server.service.ShortCircuitService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultMatcher;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith({ MockitoExtension.class })
+@WebMvcTest(controllers = { ShortCircuitParametersController.class })
+class ShortCircuitParametersControllerTest implements WithAssertions {
+    private static final String DEFAULT_PARAMETERS_JSON = "{\"predefinedParameters\":null,\"parameters\":null,\"cei909VoltageRanges\":[{\"range\":{\"comparator\":\"INSTANCE\",\"maximum\":199.99,\"minimum\":10.0,\"naturalOrdering\":true},\"rangeCoefficient\":1.1,\"voltage\":\"NaN\",\"minimumNominalVoltage\":10.0,\"maximumNominalVoltage\":199.99},{\"range\":{\"comparator\":\"INSTANCE\",\"maximum\":299.99,\"minimum\":200.0,\"naturalOrdering\":true},\"rangeCoefficient\":1.09,\"voltage\":\"NaN\",\"minimumNominalVoltage\":200.0,\"maximumNominalVoltage\":299.99},{\"range\":{\"comparator\":\"INSTANCE\",\"maximum\":500.0,\"minimum\":300.0,\"naturalOrdering\":true},\"rangeCoefficient\":1.05,\"voltage\":\"NaN\",\"minimumNominalVoltage\":300.0,\"maximumNominalVoltage\":500.0}]}";
```

Comment:
Can you save that in a resource file instead ?

Reply:
why? It's not a functionally correct object because there is not parameters (null), but it's sufficient for the test cases of this class. so I haven't create a file just for that.

Reply:
Because the string is not readable and not maintainable easily. For all our tests we just extract data in separated files in order to test them. It makes it easier to read, to understand what we test and yo modify it if necessary.